### PR TITLE
Feature/expand

### PR DESCRIPTION
**Pull request by jesperstarkar**

### CODE REVIEW

**Comment by jesperstarkar** [comment]
I want to discuss https://github.com/evs-broadcast/node-emberplus/pull/3/commits/f0166f2f453092cec034a0e20ad4a9e99d3b4bea, but to me it is the only sane way of resolving the promises recursively. I'm not 100% certain the implementation is totally correct, as I still try tro wrap my head around the mix of promises and callbacks going on.

File: client.js [hunk 1]
```diff
@@ -1,272 +1,259 @@
...
-    S101Socket.super_.call(this);
-
-    self.address = address;
-    self.port = port;
-    self.socket = null;
-    self.keepaliveInterval = 10;
-    self.codec = new S101Codec();
-    self.status = "disconnected";
-
+const EventEmitter = require('events').EventEmitter
```
Comment (dufourgilles) [inline client.js]: What's the purpose of removing all semi-colons ?

File: errors.js [hunk 2]
```diff
@@ -1,59 +1,57 @@
...
-        tagStr = "[PRIVATE " + value + "]";
-    }
-    this.message = "Unimplemented EmBER type " + tagStr;
+function UnimplementedEmberTypeError (tag) {
+  Error.captureStackTrace(this, this.constructor)
+  this.name = this.constructor.name
+  var identifier = (tag & 0xC0) >> 6
+  var value = (tag & 0x1F).toString()
+  var tagStr = tag.toString()
+  if (identifier === 0) {
```
Comment (dufourgilles) [inline errors.js]: our EVS eslint settings reject single quote.

File: device.js [hunk 3]
```diff
@@ -1,383 +1,379 @@
...
+        if (this._debug) { console.log('setValue sending ...', node.getPath(), value) }
+        self.client.sendBERNode(node.setValue(value))
+      })
     }
+  })
 }
 
-
-module.exports = {DeviceTree, DecodeBuffer};
+// function TreePath (path) {
```
Comment (dufourgilles) [inline device.js]: do not leave commented code.

**Pull request closed without merging**